apps/api/src/cart/cart.controller.ts

```diff
@@ -29,11 +29,27 @@ export class CartController {
     const cart = await this.cartService.addItemToCart(userID, newItemDto);
     return cart;
   }
+  @Delete('/') // Note: to test, apply same token decoding in Post request above
+  async removeItemFromCart(@Request() req, @Body() reqBody) {
+    const userID = await this.cartService.extractIdFromToken(req);
+    const cart = await this.cartService.removeItemFromCart(
+      userID,
+      reqBody.productID,
+    );
+    if (!cart) throw new NotFoundException('Item does not exist');
+    return { message: 'Item successfully deleted' };
+  }
 
-  @Delete('/:id') // Test delete - Accepts cart objectId
-  async deleteCart(@Param('id') userID: string) {
-    const cart = await this.cartService.deleteCart(userID);
-    if (!cart) throw new NotFoundException('Cart does not exist');
-    return cart;
+  @Delete('/:id')
+  async deleteCart(@Request() req, @Param('id') userID: string) {
+    const userIdFromToken = await this.cartService.extractIdFromToken(req);
+
+    if (userIdFromToken === userID) {
+      const cart = await this.cartService.deleteCart(userID);
+      if (!cart) throw new NotFoundException('Cart does not exist');
+      return { message: 'Cart successfully deleted' };
+    } else {
+      throw new NotFoundException('Invalid ID ');
+    }
```

Comment:
utilize token to get userID instead of passing userID in URL.

Reply:
pass cartID for Param instead of userID